californium-core/src/main/java/org/eclipse/californium/core/CoapServer.java

```diff
@@ -178,41 +180,31 @@ public CoapServer(final NetworkConfig config, final int... ports) {
 		}
 	}
 
-	/**
-	 * Sets the executor service to use for running tasks in the protocol stage.
-	 * 
-	 * @param executor The thread pool to use.
-	 * @throws IllegalStateException if this server is running and a new
-	 *             executor is provided.
-	 */
-	public void setExecutor(final ScheduledExecutorService executor) {
-		setExecutor(executor, false);
-	}
+public synchronized void setExecutors(final ScheduledExecutorService mainExecutor, final ScheduledExecutorService secondaryExecutor,  final boolean detach) {
+		if (mainExecutor == null || secondaryExecutor == null) {
+			throw new IllegalArgumentException("executors must not be null");
```

Comment:
`NullPointerException` ?

Reply:
Indent missing (source formatting)?

Reply:
About `NullPointerException` vs `IllegalArgumentException`.

I had in mind the Apaches commons lang way: https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/Validate.html (`IllegalArgument` is used for validate.notNull)

But in recent version NPE is now used : https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/Validate.html

And it seems that in californium NPE is preferred too.

So I will change it.

(I will fix formatting)